src/screens/SwitchEnvironmentScreen.tsx

```diff
@@ -1,35 +1,48 @@
 import { NativeStackScreenProps } from '@react-navigation/native-stack';
 import { useQueryClient } from '@tanstack/react-query';
 import { StatusBar } from 'expo-status-bar';
-import React from 'react';
+import React, { useState } from 'react';
 import { SafeAreaView } from 'react-native';
 
 import LoginSite from '../components/LoginSite/LoginSite';
+import LoginSiteLoader from '../components/LoginSite/LoginSiteLoader';
 import { useApi } from '../hooks/useApi';
 import { RootStackParamList } from '../navigation/types';
 import { useApiStore } from '../store/ApiStore';
 
 type Props = NativeStackScreenProps<RootStackParamList, 'SwitchEnvironment'>;
 
 export const SwitchEnvironmentScreen: React.FC<Props> = ({ navigation }) => {
+  const [isResetting, setIsResetting] = useState<boolean>(false);
   const queryClient = useQueryClient();
   const api = useApi();
   const setBaseUrlAndRefreshToken = useApiStore((state) => state.setBaseUrlAndRefreshToken);
   const onRefreshToken = async (refreshToken: string, baseUrl: string) => {
     await setBaseUrlAndRefreshToken(refreshToken, baseUrl);
-    // make sure that react query cache is cleared.
-    queryClient.resetQueries();
-    navigation.navigate('Dashboard');
+
+    // make sure that react query cache is cleared and that the new company isn't shown until all data from the
+    // previous company has been reset
+    setIsResetting(true);
+    queryClient
+      .resetQueries()
+      .finally(() => {
+        setIsResetting(false);
+        navigation.navigate('Dashboard');
+      });
```

Comment:
Gewoon await hiervoor gebruiken lijkt mij. Je kan gewoon state setten na de await weer terugzetten.

Reply:
Behalve dat als er een error optreedt tijdens `resetQueries` de gebruiker vast zit op een loading scherm en de app zelf moet force-killen. Dat is ook waarom ik dit in `.finally()` doe en niet in `.then()`.

Reply:
Je kan de await in try catch doen met een finall heb hetzelfde effect...
Je kan dan zelf de fetch weghalen.

Ik zie hier ook in de documentatie dat je errors kan negeren. door op options throwOnError op false te zetten

https://tanstack.com/query/v4/docs/react/reference/QueryClient#queryclientresetqueries

Reply:
Aangepast naar `await`. Als er een error optreedt dan wordt die netjes doorgestuurd naar de parent promise nu.

Reply:
Kan je aanwijzen welke parent promise dit afhandeld? want ik kan het niet in de commit zien.

Reply:
Deze `onRefreshToken` is zelf een async functie en returnt dus een promise.

Dus wanneer `queryClient.resetQueries()` een error throwt, wordt die nu gerethrowt en kan die worden afgevangen op de plaats waar onRefreshToken wordt aangeroepen met `onRefreshToken().catch()`, in plaats van dat die stilletjes verdwijnt naar de window.onunhandledrejection handler.